### PR TITLE
1. Migration - adding ROS enable flag

### PR DESCRIPTION
**Pull request by msvinaykumar**
## Description

- **isROSEnabled**
  - Description:  if set to True, the ROS application will use the old architecture, where ROS handles experiment creation, result updates, and recommendation updates. If set to False, ROS will utilize the Bulk API, allowing Kruize to manage experiment creation and generate recommendations on its behalf.
    Default value is false.
  - value: "false"

Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Docs update
- [ ] Breaking change (What changes might users need to make in their application due to this PR?)
- [ ] Requires DB changes

## How has this been tested?

Please describe the tests that were run to verify your changes and steps to reproduce. Please specify any test configuration required. 

- [ ] New Test X
- [x] Functional testsuite

**Test Configuration**
* Kubernetes clusters tested on: 

## Checklist :dart:

- [x] Followed coding guidelines
- [x] Comments added
- [x] Dependent changes merged
- [x] Documentation updated
- [x] Tests added or updated

## Additional information

Include any additional information such as links, test results, screenshots here

### CODE REVIEW

File: design/KruizeConfiguration.md [hunk 1]
```diff
@@ -139,3 +139,7 @@ The following environment variables are set using the `kubectl apply` command wi
     - Value: "true"
     - Details: This flag is added for getting the details of the inputs passed to the APIs and the corresponding response 
       generated by it. This helps us in debugging the API easily in case of failures.
+- **isROSEnabled**
+  - Description:  if set to True, the ROS application will use the old architecture, where ROS handles experiment creation, result updates, and recommendation updates. If set to False, ROS will utilize the Bulk API, allowing Kruize to manage experiment creation and generate recommendations on its behalf.
+    Default value is false.
+  - value: "false"
```
Comment (dinogun) [inline design/KruizeConfiguration.md:145]: This flag is not a either / or for Remote method or the local method. This flag should enable "Remote" method without having anything to do with the "Local" method.

```
**isROSEnabled**
- Description:  This flag enables the remote APIs such a updateResults and the corresponding DB Tables. If set to false, the corresponding APIs and the DB tables are not supported.
```

Comment (dinogun) [reply design/KruizeConfiguration.md:145]: This means that even if `isROSEnabled` is set to `true`, bulk APIs and corresponding DB tables are still accessible

Comment (msvinaykumar) [reply design/KruizeConfiguration.md:145]: Yes Dino .. If it is false Im planning to disable API and db table creation.

Comment (msvinaykumar) [reply design/KruizeConfiguration.md]: done

**Review by dinogun — APPROVED** [review]
LGTM

**Pull request merged**